dropshot/tests/test_config.rs

```diff
@@ -345,3 +380,361 @@ async fn test_config_bind_address_https_buffer() {
+        };
+
+        // Notify test that we're running.
+        ctx.increment_on_drop_tx.send(()).unwrap();
+
+        // Block until the test tells us to return.
+        () = ctx.block_until_rx.recv().await.unwrap();
+
+        // We weren't cancelled: mem::forget() drop_counter so its drop impl
+        // doesn't run. This leaks a reference to our drop_count, but we're in a
+        // unit test so won't worry about it.
+        mem::forget(drop_counter);
+
+        Ok(HttpResponseOk(ctx.drop_count.load(Ordering::SeqCst)))
+    }
+
+    struct ConfigBindServerHttp {
+        increment_on_drop_tx: mpsc::UnboundedSender<()>,
+        block_until_rx: async_channel::Receiver<()>,
+        drop_count: Arc<AtomicU64>,
+        drop_ran_tx: mpsc::UnboundedSender<()>,
+        log: slog::Logger,
+    }
+    impl TestConfigBindServer<hyper::client::connect::HttpConnector>
+        for ConfigBindServerHttp
+    {
+        type Context = ServerContext;
+
+        fn make_client(
+            &self,
+        ) -> hyper::Client<hyper::client::connect::HttpConnector> {
+            hyper::Client::new()
+        }
+
+        fn make_server(&self, bind_port: u16) -> HttpServer<ServerContext> {
+            let context = ServerContext {
+                drop_count: Arc::clone(&self.drop_count),
+                drop_ran_tx: self.drop_ran_tx.clone(),
+                increment_on_drop_tx: self.increment_on_drop_tx.clone(),
+                block_until_rx: self.block_until_rx.clone(),
+            };
+            let config = make_config(
+                "127.0.0.1",
+                bind_port,
+                HandlerDisposition::CancelOnDisconnect,
+            );
+            let mut api = dropshot::ApiDescription::new();
+            api.register(increment_on_drop).unwrap();
+            make_server(context, &config, &self.log, None, Some(api)).start()
+        }
+        fn make_uri(&self, bind_port: u16) -> hyper::Uri {
+            format!("http://localhost:{}/", bind_port).parse().unwrap()
+        }
+
+        fn log(&self) -> &slog::Logger {
+            &self.log
+        }
+    }
+
+    let (increment_on_drop_tx, mut increment_on_drop_rx) =
+        mpsc::unbounded_channel();
+    let (drop_ran_tx, mut drop_ran_rx) = mpsc::unbounded_channel();
+    let (block_until_tx, block_until_rx) = async_channel::unbounded();
+    let drop_count = Arc::new(AtomicU64::new(0));
+
+    let test_config = ConfigBindServerHttp {
+        increment_on_drop_tx,
+        block_until_rx,
+        drop_count: Arc::clone(&drop_count),
+        drop_ran_tx,
+        log,
+    };
+    let bind_port = 12221;
+
+    // Make sure there is not currently a server running on our expected
+    // port so that when we subsequently create a server and run it we know
+    // we're getting the one we configured.
+    let client = test_config.make_client();
+    let error = client.get(test_config.make_uri(bind_port)).await.unwrap_err();
+    assert!(error.is_connect());
+
```

Comment:
I think this stuff (and probably other stuff in this test) is not necessary here.  It's used in the tests that are actually testing binding to specific ports.  You're not testing that here, so I would specify port 0 (let the kernel decide) and leave a lot of this out.